src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs

```diff
@@ -87,7 +87,10 @@ protected override async Task HandleWebResourceRequest(CoreWebView2WebResourceRe
 				if (new Uri(AppOrigin).IsBaseOf(uri))
 				{
 					var relativePath = new Uri(AppOrigin).MakeRelativeUri(uri).ToString();
-					if (allowFallbackOnHostPage && string.IsNullOrEmpty(relativePath))
+
+					// If the path does not end in a file extension (or is empty), it's most likely referring to a page,
+					// in which case we should allow falling back on the host page.
+					if (allowFallbackOnHostPage && !Path.HasExtension(relativePath))
```

Comment:
Given this change is in the Maui Windows WebViewManager, do we have to make an equivalent change in WPF/WinForms?

Reply:
WPF and WinForms don't appear to have the bug that this PR fixes.

Reply:
Hmm that does seem a bit odd. WebView2 doesn't differ much between the platforms so I wonder what would cause this?

Reply:
@Eilon It appears the WinUI implementation is different than the implementation shared by WinForms and WPF: https://github.com/dotnet/maui/blob/0448305bc2ef0dd81aec4c7940b3860a26df8293/src/BlazorWebView/src/SharedSource/WebView2WebViewManager.cs#L222-L227
https://github.com/dotnet/maui/blob/0448305bc2ef0dd81aec4c7940b3860a26df8293/src/BlazorWebView/src/SharedSource/WebView2WebViewManager.cs#L240-L242

Since the WinUI implementation uses a `NullFileProvider`, the fallback logic in [`StaticContentProvider`](https://github.com/dotnet/aspnetcore/blob/54a7109eade9ce07eae452d4702a791638584f22/src/Components/WebView/WebView/src/StaticContentProvider.cs#L68) doesn't work as it does for WinForms and WPF. The logic in `WinUIWebViewManager` that goes through WebUI storage had more restrictive criteria than `StaticContentProvider` when deciding whether to fall back on the host page, which is the source of the bug.

The Android scenario was actually a totally different bug - `Blazor.Start()` wasn't being called when the request URI didn't precisely match the app origin (so this broke things for non-root pages).